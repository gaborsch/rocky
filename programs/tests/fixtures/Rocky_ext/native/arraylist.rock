from java, util play List, ArrayList, Arrays, Map
from java, lang play String, Long

rock X with 1, 2
let M at "one" be 1
let M at "two" be 2

say "Explicit array conversion to List<Long>"

(explicit conversion to List<Long>)
cast X into ArrList with List, Long
say toString on ArrList

cast ArrList into Y
say length of Y
say Y at 0
say Y at 1
say Y at 2


say "Implicit array conversion"
ArrList2 will be ArrayList
addAll on ArrList2 taking X
add on ArrList2 taking 3
say toString on ArrList2

cast ArrList2 into Y2
say length of Y2
say Y2 at 0
say Y2 at 1
say Y2 at 2
say Y2 at 3


say "Explicit conversion to Map<String, Long>"
cast M into MyMap with Map, String, Long
if containsKey on MyMap taking "one"
  say "Map conversion and method call OK"
else 
  say "Map conversion and method call FAILED"


