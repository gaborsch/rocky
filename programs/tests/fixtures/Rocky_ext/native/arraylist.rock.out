Explicit array conversion to List<Long>
[1, 2]
2
1
2
mysterious
Implicit array conversion
[1, 2, 3]
3
1
1
1
mysterious
Explicit conversion to Map<String, Long>
Map conversion and method call OK